geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistenceAdvisorImpl.java

```diff
@@ -533,7 +533,12 @@ public boolean checkMyStateOnMembers(Set<InternalDistributedMember> replicates)
         String message = String.format(
             "Region %s remote member %s with persistent data %s was not part of the same distributed system as the local data from %s",
             regionPath, member, remoteId, myId);
-        throw new ConflictingPersistentDataException(message);
+        replicates.remove(member);
```

Comment:
We are modifying `replicates`, which is a set that was passed into this method. I'm not quite sure what is going to happen to later logic if we have removed some of the replicates from the set that gets passed in. So I think it would be better to make a copy of replicates at the top of the method, to make sure we are not breaking other logic.

Otherwise, changes look good!

Reply:
The replicates are supposed to be adjusted here. We should not create a copy.
the whole method is to remove some of the entries from the replicates. It will remove the EQUAL ones, and offline ones, then I add the logic to remove the ones who don't know me. That means I will not GII from all these removed replicates as candidates. 
If the method removed all the replicates by other cases (such as EQUAL or offline), then I will recover from my own disk. 
As long as there's one entry in replicates, will request GII from it. 
If the last replicate is removed by this use case, should throw exception.